tokens/usecase/tokens_usecase.go

```diff
@@ -411,8 +496,18 @@ func (t *tokensUseCase) RegisterPricingStrategy(source domain.PricingSourceType,
 
 // IsValidChainDenom implements mvc.TokensUsecase.
 func (t *tokensUseCase) IsValidChainDenom(chainDenom string) bool {
-	metaData, ok := t.tokenMetadataByChainDenom[chainDenom]
-	return ok && !metaData.IsUnlisted
+	metaData, ok := t.tokenMetadataByChainDenom.Load(chainDenom)
+	if !ok {
+		return false
+	}
+
+	v, ok := metaData.(domain.Token)
+	if !ok {
+		return false
+	}
```

Comment:
Since this method starts growing in complexity, I suggest we add simple unit tests to ensure maximal test coverage while validating that the proper typing invariants are maintained.

Reply:
I have covered this method with tests: https://github.com/osmosis-labs/sqs/pull/323/files#diff-393639508320686c9dbe3457d2678cac937a6294c260d8f30a59047d394f1193R548 

Note that I am unable cover with type assertion tests because  `tokensUseCase.coingeckoIds` is unexported and tests resides in different package meaning I am unable to load it with with invalid types ( for example `int` instead of `domain.Token` ). 

There must be some reasoning behind decision making tests to live in separate package, thus I just want to learn is current coverage enough or we should cover mentioned cases as well? 

IMHO is that tests living in the same packages would simplify testing in overall and allow to cover such edge cases.

Reply:
It is possible to mock out this test case by creating a test-specific method in `export_test.go`

```go
func (t *tokensUseCase) PutInvalidTypeTokenMetadata(key string, value any) {
	t.tokenMetadataByChainDenom.Store(key, value)
}
```

Note, that since this method exists in file `..._test.go` but `usecase` package, it would only be compiled for tests but not into the prod binary.

Reply:
I just  tried  to go wtih your suggested approach and noticed that `tokensusecase.NewTokensUsecase` returns `mvc.TokensUsecase`, thus such method type is not available. If we want to go this route we can try return `TokensUseCase`  as contrete and exported type and then cast it in tests:

```
func NewTokensUsecase(tokenMetadataByChainDenom map[string]domain.Token, updateAssetsHeightInterval int, tokenRegistryLoader TokenRegistryLoader, logger log.Logger) mvc.TokensUsecase {
  ....
  return &TokensUseCase{...}
 ....
}
```

Tests:

```
usecase := tokensusecase.NewTokensUsecase(....)
usecase, ok := usecase.(tokensusecase.TokensUseCase)
```

Let me know if you favour this approach or just moving tests to the same package.